hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmGroupedPredicate.java

```diff
@@ -59,7 +59,7 @@ public boolean isNegated() {
 
 	@Override
 	public List<Expression<Boolean>> getExpressions() {
-		return Collections.singletonList( subPredicate );
+		return Arrays.asList( subPredicate );
```

Comment:
The list which `Arrays.asList` returns can not be added to unfortunately, so we need to use this:

```suggestion
		final List<Expression<Boolean>> list = new ArrayList<>( 1 );
		list.add( subPredicate );
		return list;
```